plugins/auth-backend/src/providers/oauth2-proxy/provider.ts

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import express from 'express';
+import { Logger } from 'winston';
+import { AuthenticationError } from '@backstage/errors';
+import {
+  AuthHandler,
+  SignInResolver,
+  AuthProviderFactory,
+  AuthProviderRouteHandlers,
+  AuthResponse,
+} from '../types';
+import { CatalogIdentityClient } from '../../lib/catalog';
+import { JWT } from 'jose';
+import { IdentityClient } from '../../identity';
+import { TokenIssuer } from '../../identity/types';
+import { prepareBackstageIdentityResponse } from '..';
```

Comment:
```suggestion
import { prepareBackstageIdentityResponse } from '../prepareBackstageIdentityResponse';
```

nitpick: avoid imports from all-dot directories